lib/exponent_server_sdk/parser.ex

```diff
@@ -44,11 +44,31 @@ defmodule ExponentServerSdk.Parser do
       iex> return_value
       {:ok, [%{"id" => "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX", "status" => "ok"}, %{"id" => "YYYYYYYY-YYYY-YYYY-YYYY-YYYYYYYYYYYY", "status" => "ok"}]}
   """
-  @spec parse_list(HTTPoison.Response.t()) :: success_list | error
-  def parse_list(response) do
+  @spec parse_list(HTTPoison.Response.t(), [map()] | []) :: success_list | error
+  def parse_list(response, messages \\ []) do
     handle_errors(response, fn body ->
       {:ok, json} = Poison.decode(body)
+
       json["data"]
+      |> put_missing_expo_push_token(messages)
+    end)
+  end
+
+  @spec put_missing_expo_push_token([map()], [map()] | []) :: [any()]
+  def put_missing_expo_push_token(response, messages \\ []) when is_list(response) do
+    response
+    |> Enum.with_index()
+    |> Enum.map(fn {res, index} ->
+      if res["status"] == "error" && res["details"]["error"] == "DeviceNotRegistered" &&
+           is_nil(res["details"]["expoPushToken"]) do
+        details =
+          res["details"]
+          |> Map.put("expoPushToken", Enum.at(messages, index)[:to])
```

Comment:
Are we sure that they return in the same order of how they were sent with?

Reply:
yap according this
[data will contain an array of [push tickets](https://docs.expo.dev/push-notifications/sending-notifications/#push-ticket-format) in the same order in which the messages were sent (or one push ticket object, if you send a single message to a single recipient)](https://docs.expo.dev/push-notifications/sending-notifications/#push-tickets)

Reply:
Awesome thank you for explaining 🙏